### PR TITLE
Fix newt python detection on mojave

### LINKED ISSUE #35291: Byobu - byobu-config does not work on OS X 10.14.2
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [ ] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## `brew config` output:
```
HOMEBREW_VERSION: 1.8.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 858c5e1afb0eb00079a4f014e8fbedf9065d50e4
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 1e763c8e81b8fc68b46b2a585c4b1c372581ea31
Core tap last commit: 5 hours ago
HOMEBREW_PREFIX: /usr/local
CPU: quad-core 64-bit broadwell
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.17.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.14.2-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: N/A
```
## `brew doctor` output:
```
anudeepsamaiya@orion:~$ brew doctor
Your system is ready to brew.
anudeepsamaiya@orion:~$
```

To help us debug your issue please explain:
- What you were trying to do (and why)
  I was trying to configure `byobu` using `byobu-config`
- What happened (include command output)
   The following output was thrown:
   ```
   ERROR: Could not import the python snack module
   ```
- What you expected to happen
   To have `byobu-config` screen available.
- Step-by-step reproduction instructions (by running `brew install` commands)
```
==> Installing dependencies for byobu: libpng
==> Installing byobu dependency: libpng
==> Downloading https://homebrew.bintray.com/bottles/libpng-1.6.36.mojave.bottle.tar.gz
######################################################################## 100.0%
==> Pouring libpng-1.6.36.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/libpng/1.6.36: 27 files, 1.2MB
==> Installing byobu
==> Downloading https://homebrew.bintray.com/bottles/byobu-5.127.mojave.bottle.tar.gz
Already downloaded: /Users/anudeepsamaiya/Library/Caches/Homebrew/downloads/3fe19a65f2242c25abf8e714481a8a13b67b15da5dc7b36872a6d57221c29407--byobu-5.127.mojave.bottle.tar.gz
==> Pouring byobu-5.127.mojave.bottle.tar.gz
==> Caveats
Add the following to your shell configuration file:
  export BYOBU_PREFIX=/usr/local
==> Summary
🍺  /usr/local/Cellar/byobu/5.127: 161 files, 651.5KB
==> Caveats
==> byobu
Add the following to your shell configuration file:
  export BYOBU_PREFIX=/usr/local
```
```
anudeepsamaiya@orion:~$ $BYOBU_PREFIX
bash: /usr/local: Is a directory
```
```
anudeepsamaiya@orion:~$ byobu-config
ERROR: Could not import the python snack module
anudeepsamaiya@orion:~$
```

### PR DESCRIPTION
**Pull request by martinslota**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This fixes #35291

### CODE REVIEW

**Comment by martinslota** [comment]
I only tested this on Mojave... I don't really know where system python is located on older versions.

File: Formula/newt.rb [hunk 1]
```diff
@@ -20,6 +21,8 @@ class Newt < Formula
   def install
     args = ["--prefix=#{prefix}", "--without-tcl"]
 
+    inreplace "configure", "ls /usr/include/python*/Python.h", "find /System/Library/Frameworks/Python.framework/Versions/ -name 'Python.h'"
```
Comment (fxcoudert) [inline Formula/newt.rb]: This is not the right approach. The `configure` script takes an argument after `--with-python`: see https://pagure.io/newt/c/4feae867000d6369ee036b71ad0b3ac04561bda7?branch=master

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/newt.rb (hunk 1 above)
Comment (martinslota) [reply Formula/newt.rb]: Thanks, I wasn't aware of that. However, the commit you refer to is not part of release `0.52.20`. There doesn't seem to be a newer release either:

https://pagure.io/releases/newt/

Comment (m01) [reply Formula/newt.rb]: I'm not sure what other possibilities there are..
* Lobby upstream `newt` for a new release
* Add support for building the `newt` formula from the git repo, and then use the new `--with-python` option here. I'm not sure if Homebrew's dependency management system can figure that out in a user-friendly way though?

**Pull request closed without merging**